src/python/pants/backend/java/target_types.py

```diff
@@ -80,10 +80,7 @@ class JavaTestsGeneratorSourcesField(JavaGeneratorSourcesField):
 
 class JunitTestsGeneratorTarget(TargetFilesGenerator):
     alias = "junit_tests"
-    core_fields = (
-        *COMMON_TARGET_FIELDS,
-        JavaTestsGeneratorSourcesField,
-    )
+    core_fields = (*COMMON_TARGET_FIELDS, JavaTestsGeneratorSourcesField, JunitTestTimeoutField)
```

Comment:
This should be a `moved_field` instead: the generator doesn't need to retain the timeout value after construction.